libcrux-psq/src/handshake/ciphersuite.rs

```diff
@@ -0,0 +1,747 @@
+        self.peer_longterm_mlkem_pk = Some(encapsulation_key);
+        self
+    }
+
+    #[cfg(feature = "classic-mceliece")]
+    pub fn longterm_cmc_encapsulation_key(mut self, encapsulation_key: &'a PublicKey) -> Self {
+        self.longterm_cmc_encapsulation_key = Some(encapsulation_key);
+        self
+    }
+
+    #[cfg(feature = "classic-mceliece")]
+    pub fn longterm_cmc_decapsulation_key(mut self, decapsulation_key: &'a SecretKey) -> Self {
+        self.longterm_cmc_decapsulation_key = Some(decapsulation_key);
+        self
+    }
+
+    #[cfg(feature = "classic-mceliece")]
+    pub fn peer_longterm_cmc_pk(mut self, encapsulation_key: &'a PublicKey) -> Self {
+        self.peer_longterm_cmc_pk = Some(encapsulation_key);
+        self
+    }
+
+    pub fn finish_initiator(
+        self,
+        name: CiphersuiteName,
+    ) -> Result<InitiatorCiphersuite<'a>, HandshakeError> {
+        let (peer_longterm_ecdh_pk, longterm_ecdh_keys) = self.check_common_keys_initiator()?;
+        match name {
+            CiphersuiteName::X25519ChachaPolyHkdfSha256 => {
+                Ok(InitiatorCiphersuite::X25519_ChaChaPoly_HkdfSha256(
+                    InitiatorX25519ChachaPolyHkdfSha256 {
+                        longterm_ecdh_keys,
+                        peer_longterm_ecdh_pk,
+                    },
+                ))
+            }
+            CiphersuiteName::X25519Mlkem768ChachaPolyHkdfSha256 => {
+                let Some(peer_longterm_mlkem_pk) = self.peer_longterm_mlkem_pk else {
+                    return Err(HandshakeError::BuilderState);
+                };
+                Ok(InitiatorCiphersuite::X25519_MlKem768_ChaChaPoly_HkdfSha256(
+                    InitiatorX25519Mlkem768ChachaPolyHkdfSha256 {
+                        longterm_ecdh_keys,
+                        peer_longterm_ecdh_pk,
+                        peer_longterm_mlkem_pk,
+                    },
+                ))
+            }
+            #[cfg(feature = "classic-mceliece")]
+            CiphersuiteName::X25519ClassicMcElieceChachaPolyHkdfSha256 => {
+                let Some(peer_longterm_cmc_pk) = self.peer_longterm_cmc_pk else {
+                    return Err(HandshakeError::BuilderState);
+                };
+                Ok(
+                    InitiatorCiphersuite::X25519_ClassicMcEliece_ChaChaPoly_HkdfSha256(
+                        InitiatorX25519ClassicMcElieceChachaPolyHkdfSha256 {
+                            longterm_ecdh_keys,
+                            peer_longterm_ecdh_pk,
+                            peer_longterm_cmc_pk,
+                        },
+                    ),
+                )
+            }
+            #[cfg(not(feature = "classic-mceliece"))]
+            CiphersuiteName::X25519ClassicMcElieceChachaPolyHkdfSha256 => {
+                panic!("unsupported cipersuite")
+            }
+        }
+    }
+
+    pub fn finish_responder(
+        self,
+        name: CiphersuiteName,
```

Comment:
I'd expect the name in `new`. This is fine too. I was just wondering where the ciphersuite is.